internal/core/textlayout/linebreak_simple.rs

```diff
@@ -27,13 +26,11 @@ impl<'a> Iterator for LineBreakIterator<'a> {
             let maybe_opportunity = match char {
                 '\u{2028}' | '\u{2029}' => Some(BreakOpportunity::Mandatory), // unicode line- and paragraph separators
                 '\n' => Some(BreakOpportunity::Mandatory),                    // ascii line break
-                _ if self.leading_whitespace => Some(BreakOpportunity::Allowed),
+                _ if char.is_ascii_whitespace() => Some(BreakOpportunity::Allowed),
                 _ => None,
             };
-            self.leading_whitespace = char.is_ascii_whitespace();
-
             if let Some(opportunity) = maybe_opportunity {
-                return Some((byte_offset, opportunity));
+                return Some((byte_offset + 1, opportunity));
```

Comment:
This worries me a tiny little bit, as this could return a byte offset that's out of the string bounds, if the last character is a break opportunity.

Reply:
"one past the end" is ok, not?  otherwise i can change that to not return that if this is the end.

Reply:
I think that would be good, yes.

This is a test that passes with unicode and should pass with both:

```rust
#[test]
fn test_linebreak_opportunity_at_eot() {
    let mut it = LineBreakIterator::new("Hello World\n\n");
    assert_eq!(it.next(), Some((6, BreakOpportunity::Allowed)));
    assert_eq!(it.next(), Some((12, BreakOpportunity::Mandatory)));
    assert_eq!(it.next(), None);
}
```

Reply:
I added a test that should cover this use case, does it?. The use of this function in fragments.rs seems to anyway create a break at the end if there isn't one (but the difference is that trailing_mandatory_break will be set.)

Reply:
Yes, that's fine as well. The above is a little more localised, but I think your change is good as well :)

Reply:
your test will indeed not pass with this change. But should it?

Reply:
I understand that now the linebreak_simple  and linebreak_unicode will behave differently with trailing \n,  but checking for the last codepoint is a bit of work which i *think* is unnessery since the consumer won't bother.

Unless i'm missing something

Reply:
Right now the consumer doesn't bother. But as the issue as shown, subtle differences between the two line break iterators are difficult to spot and may cause bugs that are hard to find.

So I see three options:

1. We ignore this difference for now, because the current consumers doesn't care.
2. We avoid this known difference now (but there may be more, apart from the unicode line break algorithm of course).
3. We switch to always using unicode line break algorithm, at the expense of code size.

I'm fine with either. What's your take?